legal-api/src/legal_api/models/party_role.py

```diff
@@ -150,19 +150,14 @@ def get_active_directors(legal_entity_id: int, end_date: datetime) -> list:
         return directors
 
     @staticmethod
-    def get_party_roles(legal_entity_id: int, end_date: datetime, role: str = None) -> list:
+    def get_party_roles(legal_entity_id: int, end_date: datetime = None, role: str = None) -> list:
         """Return the parties that match the filter conditions."""
-        party_roles = (
-            db.session.query(PartyRole)
-            .filter(PartyRole.legal_entity_id == legal_entity_id)
-            .filter(cast(PartyRole.appointment_date, Date) <= end_date)
-            .filter(
-                or_(
-                    PartyRole.cessation_date.is_(None),
-                    cast(PartyRole.cessation_date, Date) > end_date,
-                )
+        party_roles = db.session.query(PartyRole).filter(PartyRole.legal_entity_id == legal_entity_id)
+
+        if end_date is not None:
+            party_roles = party_roles.filter(cast(PartyRole.appointment_date, Date) <= end_date).filter(
+                or_(PartyRole.cessation_date.is_(None), cast(PartyRole.cessation_date, Date) > end_date)
```

Comment:
`PartyRole` model is essentially `EntityRole` model in legal name branch now.  

This update is good I think as it helps us keep track of changes coming in from `main`.  

But the code that will actually used and needs to be updated will be in the corresponding function in `EntityRole` model.

Reply:
Thanks for catching this Argus 🙏